### ISSUE TITLE
Redis problem with D8

### ISSUE THREAD

**Issue by macmladen** [omega8cc/boa#1451]
I've recently upgraded BOA from 3.x to 4.0.1 which has four Drupal 8 distributions, Lightning, Thunder, Open Social and Varbase.

Just for experiment, I've made a site in each on one day, but the next day I received 500 error and

`The website encountered an unexpected error. Please try again later.`

On every D8 site. I made a new one and it was working normally and the next day it failed too.

I found that in each site there is directory `.redisOn` and all those that contained that directory was not working while new ones worked until day changed (probably bitten by vampires).

If I switch to `.redisOff` site will work until tomorrow when it would be again turned to `.redisOn` and site was not working again.

I've read other posts but didn't found a solution for this.

Maybe it is of relevance that I am not running in VM but on bare metal with override directive.

**Comment by lexsoft00** [comment]
Hi @macmladen,

You would probably need to enable an alias on Aegir with a ,dev, subdomain.
Something like `mytestwebsite.dev.o1.myserver.mydomain.com` and you would be able to see what the issue is instead of `"The website encountered an unexpected error. Please try again later."`

My guess is that you probably don't have redis module enabled.

**Comment by macmladen** [comment]
@lexsoft00 I am aware of that option, however, here it is not applicable. I did try and found interesting thing: while inner links (like `example.com/user` is not working on both sites, home page of `my.dev.example.com` is working while `example.com` is **not**).

@omega8cc As I read in documentation, **Redis** does not need to be enabled, it is used anyway (`[NA] means that this module is used without the need to enable it.`). The module itself is in `o_contrib_eight` but I did not see in `settings.php`, `local.settings.php`. In `all/platform.settings.php` there are many references but are hard to follow for me.

So, all I did was just to instantiate one new **varbase** site. First, it is installed with `.redisOff` and tomorrow that changes to `.redisOn` and stops working. Just flipping it back solves the problem, however, it is not a solution, **I would like** sites to use _Redis_ to speed them up, however, for some reason it declines and it obviously have something to do with Redis.

When the state is faulty with `.redisOn`, even simple `drush status` produces following error:

```Symfony\Component\DependencyInjection\Exception\LogicException: Service 'cache_tags.invalidator.checksum' for consumer 'cache_tags.invalidator' does not implement Drupal\Core\Cache\CacheTagsInvalidatorInterface. in /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DependencyInjection/Compiler/TaggedHandlersPass.php on line 164 #0 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DependencyInjection/Compiler/TaggedHandlersPass.php(97): Drupal\Core\DependencyInjection\Compiler\TaggedHandlersPass->processServiceCollectorPass(Array, 'cache_tags.inva...', Object(Drupal\Core\DependencyInjection\ContainerBuilder))
#1 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/vendor/symfony/dependency-injection/Compiler/Compiler.php(141): Drupal\Core\DependencyInjection\Compiler\TaggedHandlersPass->process(Object(Drupal\Core\DependencyInjection\ContainerBuilder))
#2 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/vendor/symfony/dependency-injection/ContainerBuilder.php(790): Symfony\Component\DependencyInjection\Compiler\Compiler->compile(Object(Drupal\Core\DependencyInjection\ContainerBuilder))
#3 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DrupalKernel.php(1346): Symfony\Component\DependencyInjection\ContainerBuilder->compile()
#4 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DrupalKernel.php(920): Drupal\Core\DrupalKernel->compileContainer()
#5 /opt/tools/drush/8/drush/lib/Drush/Drupal/DrupalKernel.php(61): Drupal\Core\DrupalKernel->initializeContainer()
#6 /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DrupalKernel.php(476): Drush\Drupal\DrupalKernel->initializeContainer()
#7 /opt/tools/drush/8/drush/lib/Drush/Boot/DrupalBoot8.php(146): Drupal\Core\DrupalKernel->boot()
#8 /opt/tools/drush/8/drush/includes/bootstrap.inc(354): Drush\Boot\DrupalBoot8->bootstrap_drupal_full()
#9 /opt/tools/drush/8/drush/includes/bootstrap.inc(473): drush_bootstrap(5, 6)
#10 /opt/tools/drush/8/drush/lib/Drush/Boot/BaseBoot.php(59): drush_bootstrap_to_phase(6)
#11 /opt/tools/drush/8/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#12 /opt/tools/drush/8/drush/drush.php(12): drush_main()
#13 {main}
Drush command terminated abnormally due to an unrecoverable error.                                                                                                                                                                                                                                          [error]
Symfony\Component\DependencyInjection\Exception\LogicException: Service &#039;cache_tags.invalidator.checksum&#039; for consumer &#039;cache_tags.invalidator&#039; does not implement Drupal\Core\Cache\CacheTagsInvalidatorInterface. in Drupal\Core\DependencyInjection\Compiler\TaggedHandlersPass->processServiceCollectorPass() (line 164 of /data/disk/o1/distro/012/varbase-8.6.8-8.7.0/docroot/core/lib/Drupal/Core/DependencyInjection/Compiler/TaggedHandlersPass.php).
```

**Comment by omega8cc** [comment]
Redis must be enabled on D8 (unlike in D7 and D6), but BOA head comes with a workaround, which makes it work even without Redis enabled (it will be enabled automatically next night), so redisOff and redisOn are no longer used for anything. Please upgrade both barracuda and octopus to head to receive all fixes before reporting bugs affecting Drupal newer versions. We are going to release new BOA stable before 2019 end.

**Comment by macmladen** [comment]
@omega8cc @lexsoft00 You were right, **redis** module has to be _enabled_ I was assuming it is by _default_ as it was the case earlier with Drupal 7.

I confirm that **enabling redis** fixes both included and custom platforms.

Once again, _assuming_ leads only to errors.

**Comment by omega8cc** [comment]
@macmladen Just to clarify, in D7 and D6 the integration module was never enabled in the strict sense, but worked just fine anyway, while the requirement to actually enable it like any other module was introduced due to the way D8 core works, but since enabling it during installation may cause serious problems (again due to known issues related to handling caches on Drupal install), we have initially left it disabled to avoid conflicts and enable only the next day. It was certainly confusing and problematic but thanks to workarounds in BOA head should no longer be a problem.

**Comment by omega8cc** [comment]
Just to clarify, we have fixed this in September 2019.. so Redis now works even if it's not enabled yet 2b44f89ae92e0f67d865c010704b721094fee095
